File: src/RentalForge.Api/Data/Entities/Film.cs
using NpgsqlTypes;

namespace RentalForge.Api.Data.Entities;

/// <summary>
/// Represents a film in the catalog. Related to actors and categories through join tables.
/// </summary>
public class Film
{
    public int FilmId { get; set; }
    public string Title { get; set; } = null!;
    public string? Description { get; set; }
    public int? ReleaseYear { get; set; }
    public int LanguageId { get; set; }
    public int? OriginalLanguageId { get; set; }
    public short RentalDuration { get; set; }
    public decimal RentalRate { get; set; }
    public short? Length { get; set; }
    public decimal ReplacementCost { get; set; }
    public MpaaRating? Rating { get; set; }
    public DateTime LastUpdate { get; set; }
    public string[]? SpecialFeatures { get; set; }
    public NpgsqlTsVector Fulltext { get; set; } = null!;

    public Language Language { get; set; } = null!;
    public Language? OriginalLanguage { get; set; }
    public ICollection<FilmActor> FilmActors { get; set; } = [];
    public ICollection<FilmCategory> FilmCategories { get; set; } = [];
    public ICollection<Inventory> Inventories { get; set; } = [];
}

File: src/RentalForge.Web/src/types/film.ts
export type MpaaRating = 'G' | 'PG' | 'PG-13' | 'R' | 'NC-17';

export interface FilmListItem {
  id: number;
  title: string;
  description: string | null;
  releaseYear: number | null;
  languageId: number;
  originalLanguageId: number | null;
  rentalDuration: number;
  rentalRate: number;
  length: number | null;
  replacementCost: number;
  rating: MpaaRating | null;
  specialFeatures: string[] | null;
  lastUpdate: string;
}

export interface FilmDetail extends FilmListItem {
  languageName: string;
  originalLanguageName: string | null;
  actors: string[];
  categories: string[];
}

export interface FilmSearchParams {
  search?: string;
  category?: string;
  rating?: MpaaRating;
  yearFrom?: number;
  yearTo?: number;
  page?: number;
  pageSize?: number;
}
